vars/artifactSetVersion.groovy

```diff
@@ -28,11 +28,18 @@ enum GitPushMode {NONE, HTTPS, SSH}
      */
     'buildTool',
     /**
-     * Controls if the changed version is committed and pushed to the git repository.
-     * If this is enabled (which is the default), you need to provide `gitCredentialsId` and `gitSshUrl`.
+     * Controls if the changed version is committed.
      * @possibleValues `true`, `false`
      */
     'commitVersion',
+    /**
+     * Controls if a tag is created.
+     */
+    'createTag',
+    /**
+     * Controls if a version is created.
+     */
+    'createVersion',
```

Comment:
Why would  I call  a setVersion command if I did not want to create a version. I guess we should properly evaluate the use case here.

That really sounds strange to me ...  ‘createVersion: false’

Reply:
Of course triggering a new version will always be part of the game. Using this step without creating a version is pointless. It is about calling the step multiple times in several modes:

1.) create a new version (two behaviors: with creating a commit / without creating a commit)
2.) create and tag and push to upstream.

The approach here is to create a new version early in the build since the version information is incorporated into the artifacts. But we might want to publish a corresponding tag later in the build only if the build, maybe together with some tests, was successful. Or we want to publish the tag only after some deployment has happened.

- We remember a new version in the CPE anyway. We can hide the `createVersion` from the outside since we can derive that from the CPE: If there is already a new version we don't create a second version.
- We can also introduce a second step for publishing that tag in case we don't want to call the same step twice in different modes.
- We can also agree that this is too far away / too weird for our standard piper-lib use-cases. In this case we can:
  - provide something like that on a per-customer basis.
  - decide to create another shared lib in another repo having some kind of add-on steps. With that approach we should be very, very careful. That would be some kind of a last resort. IMO that would only be an option if we really don't want to have that in the piper lib and we (or consulting) have the impression we need that again and again for different customers.

@OliverNocon : which way would you prefer?

Reply:
what about using `githubPublishRelease` for the tagging part. This is actually doing this plus some additional nice stuff.

For non GitHub projects one could think about adding a new step `gitTag...`